captum/optim/models/_common.py

```diff
@@ -21,9 +21,13 @@ def get_layers(net: nn.Module, prefix: List = []) -> None:
             for name, layer in net._modules.items():

```

Comment:
btw, do we assume every module has a name? What if the name is `None`? do we want to handle it like `layer1.None.layer3`

Reply:
@aobo-y Looking at the `nn.Module` class, using `None` for the name does not appear to be possible: https://github.com/pytorch/pytorch/blob/master/torch/nn/modules/module.py#L370

The variable has to be `torch._six.string_classes`, and one cannot do `None = nn.Module()` for initial creation of the module.

Reply:
@ProGamerGov then we should use `named_children()` https://pytorch.org/docs/stable/generated/torch.nn.Module.html#torch.nn.Module.named_children

This is the public api to get the immediate children. Underneath, it does the same as your implementation https://github.com/pytorch/pytorch/blob/master/torch/nn/modules/module.py#L1700 , but we can avoid accessing the private property while no need to check `layer is None`

Reply:
Okay, I've replaced `net._modules.items()` with `net.named_children()`! 

I think the usage of the private property might have been a holdover from back before the `named_children()` function existed.